packages/blitz-auth/src/client/index.tsx

```diff
@@ -245,79 +245,95 @@ export function getAuthValues<TProps = any>(
+        if (publicData.userId) {
+          debug("[BlitzAuthInnerRoot] logged in")
 
-          debug("[BlitzAuthInnerRoot] redirecting to", redirectUrl)
-          const error = new RedirectError(redirectUrl)
-          error.stack = null!
-          throw error
-        }
-      } else {
-        debug("[BlitzAuthInnerRoot] logged out")
-        if (authenticate && typeof authenticate === "object" && authenticate.redirectTo) {
-          let {redirectTo} = authenticate
-          if (typeof redirectTo !== "string") {
-            redirectTo = formatWithValidation(redirectTo)
+          if (typeof redirectAuthenticatedTo === "function") {
+            redirectAuthenticatedTo = redirectAuthenticatedTo({
+              session: publicData,
+            })
           }
 
-          const url = new URL(redirectTo, window.location.href)
-          url.searchParams.append("next", window.location.pathname)
-          debug("[BlitzAuthInnerRoot] redirecting to", url.toString())
-          const error = new RedirectError(url.toString())
-          error.stack = null!
-          throw error
+          if (redirectAuthenticatedTo) {
+            const redirectUrl =
+              typeof redirectAuthenticatedTo === "string"
+                ? redirectAuthenticatedTo
+                : formatWithValidation(redirectAuthenticatedTo)
+
+            debug("[BlitzAuthInnerRoot] redirecting to", redirectUrl)
+            const error = new RedirectError(redirectUrl)
+            error.stack = null!
+            throw error
+          }
+        } else {
+          debug("[BlitzAuthInnerRoot] logged out")
+          if (authenticate && typeof authenticate === "object" && authenticate.redirectTo) {
+            let {redirectTo} = authenticate
+            if (typeof redirectTo !== "string") {
+              redirectTo = formatWithValidation(redirectTo)
+            }
+
+            const url = new URL(redirectTo, window.location.href)
+            url.searchParams.append("next", window.location.pathname)
+            debug("[BlitzAuthInnerRoot] redirecting to", url.toString())
+            const error = new RedirectError(url.toString())
+            error.stack = null!
+            throw error
+          }
         }
       }
+
+      return <Page {...props} />
     }
 
-    return <Page {...props} />
-  }
+    for (let [key, value] of Object.entries(Page)) {
+      // @ts-ignore
+      AuthRoot[key] = value
+    }
+    if (process.env.NODE_ENV !== "production") {
+      AuthRoot.displayName = `BlitzAuthInnerRoot`
+    }
 
-  for (let [key, value] of Object.entries(Page)) {
-    // @ts-ignore
-    AuthRoot[key] = value
-  }
-  if (process.env.NODE_ENV !== "production") {
-    AuthRoot.displayName = `BlitzAuthInnerRoot`
+    return AuthRoot
   }
-
-  return AuthRoot
 }
 
 export interface AuthPluginClientOptions {
   cookiePrefix: string
+  defaults?: AuthPluginDefaultOptions
```

Comment:
I think this will be a lot more clear what it is when reading the code
```suggestion
  pageDefaults?: AuthPluginDefaultOptions
```